struct UserProfile {
    1: required string user_id
    2: required string username
    3: optional string email
}